Tests/IntegrationTests/SwiftKafkaTests.swift

```diff
@@ -0,0 +1,222 @@
+        let consumer = try await KafkaConsumer(
+            topic: topic,
+            partition: KafkaPartition(rawValue: 0),
+            config: config,
+            logger: .kafkaTest
+        )
+
+        let testMessage = KafkaProducerMessage(
+            topic: topic,
+            key: "key",
+            value: "Hello, World! \(Date().description)"
+        )
+
+        try await Self.resetConsumerOffset(consumer)
+        try await Self.sendAndAcknowledgeMessages(producer: producer, messages: [testMessage])
+
+        var consumedMessage: KafkaConsumerMessage?
+        for await messageResult in consumer.messages {
+            guard case .success(let message) = messageResult else {
+                continue
+            }
+            consumedMessage = message
+            break
+        }
+
+        XCTAssertEqual(testMessage.topic, consumedMessage?.topic)
+        XCTAssertEqual(testMessage.key, consumedMessage?.key)
+        XCTAssertEqual(testMessage.value, consumedMessage?.value)
+
+        try await consumer.close()
+        try await producer.shutdownGracefully()
+    }
+
+    func testProduceAndConsumeWithCommitSync() async throws {
+        let topic = "commit-sync-test-topic"
+
+        try config.set("false", forKey: "enable.auto.commit")
+
+        let producer = try await KafkaProducer(config: config, logger: .kafkaTest)
+        let consumer = try await KafkaConsumer(
+            topics: [topic], // TODO: multiple topics
+            groupID: "commit-sync-test-group-id",
+            config: config,
+            logger: .kafkaTest
+        )
+
+        let testMessages = Array(0...9).map {
+            KafkaProducerMessage(
+                topic: topic,
+                key: "key",
+                value: "Hello, World! \($0) - \(Date().description)"
+            )
+        }
+
+        try await Self.resetConsumerOffset(consumer)
+        try await Self.sendAndAcknowledgeMessages(producer: producer, messages: testMessages)
+
+        var consumedMessages = [KafkaConsumerMessage]()
+        for await messageResult in consumer.messages {
+            guard case .success(let message) = messageResult else {
+                continue
+            }
+            consumedMessages.append(message)
+            try await consumer.commitSync(message)
+
+            if consumedMessages.count >= testMessages.count {
+                break
+            }
+        }
+
+        XCTAssertEqual(testMessages.count, consumedMessages.count)
+
+        try await consumer.close()
+        try await producer.shutdownGracefully()
+    }
+
+    // TODO: also test concurrently?
+
+    // MARK: - Helpers
+
+    private static func resetConsumerOffset(_ consumer: KafkaConsumer) async throws {
+        let start = Date()
+        let timeout = 10.0
+        while Date() < start + timeout {
+            _ = try await consumer.poll(timeout: 1000) // TODO: poll needs to be private, do something else here
```

Comment:
Ideally, we would be able to make `poll()` `private` and have an `AsyncSequence` with a **timeout** here. Though afaik, there is no such thing so we would have to implement it ourselves.

What do you think of this approach?

https://forums.swift.org/t/running-an-async-task-with-a-timeout/49733